### PR TITLE
feat(tools): enum, regex, referential type feature for csharp codegen

### PR DESCRIPTION
**Pull request by ragi-dayananda**
### Changes
 - Add regex annotation for csharp codegen when regex configured for a field
 - UUID nullable type bug fix
 - Make optional string properties as nullable string. 
 - Custom enum value support with AcceptedValue decorator
 - Handle referential type (-->) in csharp codegen

### Author Checklist
- [x] Ensure you provide a [DCO sign-off](https://github.com/probot/dco#how-it-works) for your commits using the `--signoff` option of git commit.
- [ ] Vital features and changes captured in unit and/or integration tests
- [ ] Commits messages follow [AP format](https://github.com/accordproject/techdocs/blob/master/DEVELOPERS.md#commit-message-format)
- [ ] Extend the documentation, if necessary
- [ ] Merging to `main` from `fork:branchname`

### CODE REVIEW

File: packages/concerto-tools/lib/codegen/fromcto/csharp/csharpvisitor.js [hunk 1]
```diff
@@ -283,8 +284,17 @@ class CSharpVisitor {
 
         let fieldType = externalFieldType ? externalFieldType : this.getFieldType(field);
 
+        if (fieldType === 'String') {
```
Comment (dselman) [inline packages/concerto-tools/lib/codegen/fromcto/csharp/csharpvisitor.js:287]: I worry that we might be using an `externalFieldType` here...

File: packages/concerto-tools/lib/codegen/fromcto/csharp/csharpvisitor.js [hunk 2]
```diff
@@ -283,8 +284,17 @@ class CSharpVisitor {
...
+            let validator = field.getValidator();
```
Comment (dselman) [inline packages/concerto-tools/lib/codegen/fromcto/csharp/csharpvisitor.js:288]: can be `const`?

File: packages/concerto-tools/lib/codegen/fromcto/csharp/csharpvisitor.js (hunk 1 above)
Comment (ragi-dayananda) [reply packages/concerto-tools/lib/codegen/fromcto/csharp/csharpvisitor.js:290]: yes, we might. But external type can be UUID scalar type. We don't want to add regex decorator for UUID.

**Review by dselman — APPROVED** [review]
Looks great!

**Pull request merged**